mrbgems/mruby-io/test/mruby_io_test.c

```diff
@@ -80,6 +113,16 @@ mrb_io_test_io_setup(mrb_state *mrb, mrb_value self)
 #if !defined(_WIN32) && !defined(_WIN64)
   int fd2, fd3;
   struct sockaddr_un sun0;
+
+  if(!(socket_available_p = mrb_io_socket_available())) {
+    char *tmpdir;
+    wd_save = open(".", O_DIRECTORY);
+    tmpdir = getenv("TMPDIR");
+    if (tmpdir)
+      assert(!chdir(tmpdir));
+    else
+      assert(!chdir("/tmp"));
+  }
```

Comment:
I think the side effect in assertions are not good because it changes the behavior depending on whether it is in debug mode or not.

In the first place, it's better not to change the directory if possible, but at least it should go back to the original directory, I think.

FYI: It may be better to use `mrb_assert()` instead of `assert()`.

Reply:
> In the first place, it's better not to change the directory if possible, but at least it should go back to the original directory, I think.

I'm sorry, I misunderstood about the following, so please ignore the quoted comments.
* Moves to a temporary directory when some tests don't work in the current directory.
* Returns to the original directory via `mrb_io_test_io_cleanup()`.

Reply:
Yeah, `assert(3)` is ZATSU... I expected that I got some review

Reply:
`mrb_assert()` looks better